app/backend/src/service/keycloakService.ts

```diff
@@ -0,0 +1,158 @@
+import { Axios } from 'axios';
+import { Response, NextFunction } from 'express-serve-static-core';
+import di from '../di';
+import { RawToken, UserAuthenticated, UserKeycloak } from '../model/AuthKeycloak';
+import { UserLoginInput } from '../model/User';
+import { DotConfig } from '../util/config/Config';
+const querystring = require('querystring');
+
+export class KeycloakService {
+    private _axios: Axios;
+    private readonly _env: DotConfig;
+    private _adminToken: RawToken = null;
+    private _adminTokenExpiration = 0;
+
+    constructor(axios: Axios, env: DotConfig) {
+        this._axios = axios;
+        this._env = env;
+    }
+    async createUser(
+        firstName: string,
+        lastName: string,
+        email: string,
+        enabled: boolean,
+        password?: string,
+    ): Promise<UserKeycloak> {
+        const adminToken = await this.fetchValidAdminToken();
+
+        const form = {
+            firstName,
+            lastName,
+            username: email,
+            email: email,
+            enabled: enabled ? 'true' : 'false',
+            credentials: password
+                ? [
+                      {
+                          type: 'password',
+                          value: password,
+                      },
+                  ]
+                : [],
+        };
+        const headers = { Authorization: `Bearer ${adminToken}`, 'Content-Type': `application/json` };
+        const url = `${di.env.KEYCLOAK_AUTH_URL}admin/realms/${di.env.KEYCLOAK_REALM}/users`;
+        let response;
```

Comment:
if this response variable is being used one time for each request, its a better practice to use const. Otherwise, nvm